### ISSUE TITLE
Experiment: promoters from different evolutionary timescales

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#55]
## Description

Following the general ideas of #6 , train models on promoters from different evolutionary timescales and check downstream task performance.

## Hypothesis or Goal

Understand tradeoffs between data quantity and evolutionary relevance.

Technical details:
- Make sure to train for at least 1 epoch (17k steps will cover the largest animal dataset)

## Links

[Training code](https://github.com/marin-community/marin/blob/73dbb49e319927869a9b16684545b62fde9b6906/experiments/dna/exp55_evolutionary_timescales.py)
[Analysis code](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1)
Wandb: [humans](https://wandb.ai/gonzalobenegas/marin/runs/exp55-humans-r01-0fa9f3?nw=nwusergonzalobenegas), [primates](https://wandb.ai/gonzalobenegas/marin/runs/exp55-primates-r01-70b1a5?nw=nwusergonzalobenegas), [mammals](https://wandb.ai/gonzalobenegas/marin/runs/exp55-mammals-r01-52f413?nw=nwusergonzalobenegas), [vertebrates](https://wandb.ai/gonzalobenegas/marin/runs/exp55-vertebrates-r01-64b0da?nw=nwusergonzalobenegas), [animals](https://wandb.ai/gonzalobenegas/marin/runs/exp55-animals-r01-c5dd2f?nw=nwusergonzalobenegas)

## Results

Eval: TraitGym Mendelian traits v2 (validation set, only odd chromosomes).
- Model trained on mammals reaches a good performance faster.

<img width="1083" height="386" alt="Image" src="https://github.com/user-attachments/assets/20d11518-2f58-4c21-aeba-4711c52b7275" />

### Language modeling

<img width="500" height="500" alt="Image" src="https://github.com/user-attachments/assets/fb76ffc3-012f-4586-a328-832a0822a9f2" />
<img width="500" height="500" alt="Image" src="https://github.com/user-attachments/assets/3a1d4413-ff71-4900-8c1d-059d5619974a" />

- There's something cyclic related to how learning rate is cycled every 1k steps. However, within the primates you also see some recurring patterns *within* a cycle? One epoch is 447 steps for the primate dataset. Could this mean that in each cycle the data is restarted, and seen in the same order? Or is this normal?
<img width="500" height="304" alt="Image" src="https://github.com/user-attachments/assets/78fc777a-6dd7-4c32-bfc0-d12e88fdd491" />

- Validation loss (log scale) suggests that humans and primates models overfit. The other models could also be overfitting (just not as much) but we don't realize because *no sequence similarity filter has been applied* #28 (not something people have implemented in DNA AFAIK).
- Note: the validation set is the same in all 5 models (only human promoters)  
<img width="500" height="500" alt="Image" src="https://github.com/user-attachments/assets/2dad1961-094a-4379-be6a-b9c0267a7c1e" />

## Follow up questions
- What happens as we train for longer?
- What happens as we vary model size?
- What happens on other regions of the genome?

**Comment by gonzalobenegas** [comment]
🤖 Raw AUPRC numbers behind the plot above, assembled from the [evals_v1 pipeline outputs](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1) in S3 (`s3://oa-bolinas/snakemake/analysis/evals_v1/results/metrics/traitgym_mendelian_v2/exp55-*/`).

**File:** [exp55_auprc_traitgym_mendelian_v2.parquet](https://gist.githubusercontent.com/gonzalobenegas/34dded27de006485839006f8349980be/raw/d0219916dace8cad13d044fc137a5409bd5f8923/exp55_auprc_traitgym_mendelian_v2.parquet)

**Shape:** 3060 rows = 5 models × 17 checkpoint steps × 9 subsets × 4 score_types.

**Columns:**
- `model` — `humans` / `primates` / `mammals` / `vertebrates` / `animals`
- `model_run` — full training-run name (e.g. `exp55-humans-r01-0fa9f3`)
- `step` — checkpoint step
- `metric` — `AUPRC` (the only metric computed for this dataset)
- `score_type` — `minus_llr`, `abs_llr`, `embed_last_l2`, `embed_middle_l2`
- `subset` — `global`, `tss_proximal`, `5_prime_UTR_variant`, `3_prime_UTR_variant`, `non_coding_transcript_exon_variant`, `missense_variant`, `synonymous_variant`, `splicing`, `distal`
- `value` — AUPRC
- `n_pos`, `n_neg` — positives / negatives in that subset

Eval dataset: [`bolinas-dna/evals-traitgym_mendelian_v2`](https://huggingface.co/datasets/bolinas-dna/evals-traitgym_mendelian_v2) (validation set, odd chromosomes). The plot in the issue uses `score_type == 'minus_llr'` and `subset in {'tss_proximal', '5_prime_UTR_variant'}`.

```python
import pandas as pd
df = pd.read_parquet('https://gist.githubusercontent.com/gonzalobenegas/34dded27de006485839006f8349980be/raw/d0219916dace8cad13d044fc137a5409bd5f8923/exp55_auprc_traitgym_mendelian_v2.parquet')
```

**Comment by gonzalobenegas** [comment]
🤖 Updated plot using the post-#194 evals_v2 dataset (`mendelian_traits`, hf_revision `4aed58e`).

![exp55_evolutionary_timescales_auprc](https://gist.githubusercontent.com/gonzalobenegas/3649e68fb63ca1f3443e4486078eb4d8/raw/c785daa5a1a938365ace6861791fa3e978ac53cf/exp55_evolutionary_timescales_auprc.svg)

Recipe: [`plots/exp55_evolutionary_timescales.py`](https://github.com/Open-Athena/marin-dna/blob/0058ec4779401307a9b3a3c9218127511711bce4/plots/exp55_evolutionary_timescales.py)
